tests/test_dataset.py

```diff
@@ -28,35 +28,35 @@
 
 import lsst.utils.tests
 import lsst.pex.exceptions as pexExcept
+from lsst.ap.verify.config import Config
 from lsst.ap.verify.dataset import Dataset
 
 
 class DatasetTestSuite(lsst.utils.tests.TestCase):
 
     @classmethod
     def setUpClass(cls):
-        cls.testDataset = 'ap_verify_hits2015'
-        cls.datasetKey = 'HiTS2015'
-        cls.obsPackage = 'obs_decam'
-        cls.camera = 'decam'
+        cls.testDataset = 'ap_verify_testdata'
+        cls.datasetKey = 'test'
+        cls.obsPackage = 'obs_test'
+        cls.camera = 'test'
         try:
             lsst.utils.getPackageDir(cls.testDataset)
         except pexExcept.NotFoundError:
             raise unittest.SkipTest(cls.testDataset + ' not set up')
 
+        # Hack the config for testing purposes
+        # Note that Config.instance is supposed to be immutable, so some code may break
+        Config.instance._allInfo['datasets.' + cls.datasetKey] = cls.testDataset
```

Comment:
Your comments here don't exactly inspire confidence. What's "some code may break" all about?

Reply:
`Config.instance` loads itself from `ap_verify`'s config files before it gets used (at import time, I think?). The assumption is that `Config.instance` simply reflects the contents of those files, i.e., it does not change within a program run.

So you could imagine that some code (yet to be written) reads something from `Config.instance` before `DatasetTestSuite.setUpClass` runs, then `Config.instance` gets modified, then the code reads `Config.instance` later and gets an inconsistent result, which causes it to do something crazy (kind of literally). This could be code in or called by a different test, so it's not possible to control it.

The alternative is actually putting `ap_verify_testdata` in the config. However, it would then be usable by `ap_verify` itself, and even show up as one of the available datasets in the `-h` text. A hack that only affects the tests but not the program itself seems like the lesser evil.

Reply:
Huh. That is a somewhat interesting hypothetical case! I didn't have the context that `Config.instance` isn't "normally" accessed from a place like this. Perhaps instead of "so some code may break" you could say "so if the config has changed since import there may be a discrepancy."

Reply:
Not sure what you mean by "normally accessed" -- any part of the `ap_verify` codebase is free to access `Config.instance` at will; that's what it's for.

How about:
> Note that Config.instance is supposed to be immutable, so, depending on initialization order, this modification may cause other tests to see inconsistent config values